temporian/implementation/numpy_cc/operators/window.cc

```diff
@@ -38,18 +38,36 @@ py::array_t<OUTPUT> accumulate(const ArrayD &evset_timestamps,
   size_t begin_idx = 0;
 
   for (size_t sampling_idx = 0; sampling_idx < n_event; sampling_idx++) {
+    // Note: We accumulate values in (t-window_length, t] with t=
+    // v_timestamps[sampling_idx], and there may be several contiguous equal
+    // values in v_timestamps.
 
+    // Add all values with same timestamp as the current one.
     accumulator.Add(v_values[sampling_idx]);
+    size_t same_ts_idx = sampling_idx;

```

Comment:
I think the code below could be simpler if you store "sampling_idx+1" instead of "sampling_idx".

- No need for the +1 in the while loop.
- Need to do same_ts_idx++ after the .Add (no change in complexity).
- < instead of <= in for loop on line 65.
- Can replace the for loop on line 40 with a while loop (or a for loop without the "sampling_idx++" part).

Reply:
loved it!